packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -99,14 +97,29 @@ export function FileUploader({
   const progressCallback = useCallback(
     (index: number) => {
       return (progress: { loaded: number; total: number }) => {
-        const percentage = Math.floor((progress.loaded / progress.total) * 100);
-        const status = fileStatusesRef.current[index];
-        fileStatusesRef.current[index] =
-          percentage !== 100
-            ? { ...status, percentage, fileState: 'loading' }
-            : { ...status, percentage, fileState: 'success' };
-        const newFileStatuses = [...fileStatusesRef.current];
-        setFileStatuses(newFileStatuses);
+        setFileStatuses((prevFileStatuses) => {
+          const prevStatus = { ...prevFileStatuses[index] };
+
+          const progressPercentage = Math.floor(
+            (progress.loaded / progress.total) * 100
+          );
+          const updatedStatus =
+            progressPercentage !== 100
+              ? {
+                  ...prevStatus,
+                  percentage: progressPercentage,
+                  fileState: 'loading' as FileState,
+                }
+              : {
+                  ...prevStatus,
+                  percentage: progressPercentage,
+                  fileState: 'success' as FileState,
+                };
```

Comment:
Could just be:
```
        const fileState = progressPercentage !== 100 ? 'loading' : 'success'
        const updatedStatus = {
          ...prevStatus,
          percentage: progressPercentage,
          fileState as FileState,
        }

Reply:
👍

Reply:
[b6c93a3](https://github.com/aws-amplify/amplify-ui/pull/3028/commits/b6c93a3be7ac04605ecb0f92846fdbaee8b465bc)